internal/pkg/cli/task_run.go

```diff
@@ -954,6 +954,15 @@ func (o *runTaskOpts) deploy() error {
 		deployOpts = []awscloudformation.StackOption{awscloudformation.WithRoleARN(o.targetEnvironment.ExecutionRoleARN)}
 	}
 
+	var boundaryPolicy string
+	if o.appName != "" {
+		app, err := o.store.GetApplication(o.appName)
```

Comment:
nit: what do you think of caching the `app config`  [here](https://github.com/aws/copilot-cli/blob/a6ae6784bf9a46cecf7697e4e33857ba5c3370a7/internal/pkg/cli/task_run.go#L1049) and try to reuse the cached version here?

Reply:
Yeah, I thought of that, but then we won't have it if they use the selector, not the flag to indicate the app name.